### PR TITLE
Make M1 tests green

### PR DESCRIPTION
**Pull request by albanD**
This is skipping all the failing tests and add a new master job to test on M1

### CODE REVIEW

**Review by seemethere — APPROVED** [review]

File: test/test_cpp_extensions_jit.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def remove_build_path():
     if os.path.exists(default_build_root):
         shutil.rmtree(default_build_root)
 
-
+@unittest.skipIf(IS_ARM64, "Does not work on arm")
```
Comment (soulitzer) [inline test/test_cpp_extensions_jit.py:40]: Might conflict with https://github.com/pytorch/pytorch/pull/82117

**Review by soulitzer — APPROVED** [review]

File: test/test_cpp_extensions_jit.py (hunk 1 above)
Comment (albanD) [reply test/test_cpp_extensions_jit.py:40]: Yes it does, will rebase once the CI job is fixed!

**Review by malfet — APPROVED** [review]

File: .jenkins/pytorch/macos-test.sh [hunk 2]
```diff
@@ -5,7 +5,11 @@
 source "$(dirname "${BASH_SOURCE[0]}")/macos-common.sh"
 
 conda install -y six
-pip install -q hypothesis "expecttest==0.1.3" "librosa>=0.6.2" "numba<=0.49.1" psutil "scipy==1.6.3"
+if [[ ${BUILD_ENVIRONMENT} = *arm64* ]]; then
+  pip install -q hypothesis "expecttest==0.1.3" "librosa>=0.6.2" numba psutil scipy
```
Comment (albanD) [inline .jenkins/pytorch/macos-test.sh:9]: TODO: re-pin numba and scipy once I know which version is used in CI

**Review by seemethere — APPROVED** [review]

**Comment by albanD** [comment]
Everything is re-pinned and rebased now. Will merge once CI is green!

**Pull request closed without merging**